packages/arb-token-bridge-ui/src/hooks/useTransactionHistory.ts

```diff
@@ -568,10 +601,20 @@ export const useTransactionHistory = (
       const startIndex = _page * MAX_BATCH_SIZE
       const endIndex = startIndex + MAX_BATCH_SIZE
 
-      return Promise.all(
-        dedupedTransactions
-          .slice(startIndex, endIndex)
-          .map(transformTransaction)
+      const transactionsBatch = dedupedTransactions.slice(startIndex, endIndex)
```

Comment:
Only moved batched `dedupedTransactions` to its own variable `transactionsBatch` so we can reuse it in failed chains in the catch block.